typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -579,6 +579,17 @@
   [#:frees (λ (f) (make-invariant (f value)))]
   [#:key 'continuation-mark-key])
 
+;; Newtype
+;; comes from define-new-subtype
+;; nm: a symbol representing the name of the Newtype
+;; id: a symbol created with gensym
+;; ty: a type for the representation
+(def-type Newtype ([nm symbol?] [id symbol?] [ty Type/c])
```

Comment:
@johanatan It is, precisely, a subtype, so that part of the name make sense.

@rfindler It's "new" in the sense of "fresh" -- different definitions produce unrelated types, unlike `define-type`, for example.